src/components/ImageSlider.tsx

```diff
@@ -8,34 +8,25 @@ type ImageSliderProps = {
 
 const ImageSlider = ({ urls }: ImageSliderProps) => {
   const [currentIndex, setCurrentIndex] = useState(0);
-  const [isHovering, setIsHovering] = useState(false);
-  const [translatePercent, setTranslatePercent] = useState(0);
-  const ulListRef = useRef<HTMLUListElement>(null);
-  const firstImageRef = useRef<HTMLLIElement>(null);
-
-  const calculateTranslatePercent = () => {
-    if (ulListRef.current && firstImageRef.current) {
-      const ulListWidth = ulListRef.current.clientWidth - 20;
-      const firstImageWidth = firstImageRef.current.clientWidth;
-      const imageRatio = (firstImageWidth / ulListWidth) * 100;
-      setTranslatePercent(imageRatio);
-    }
-  };
+  const [showControls, setShowControls] = useState(false);
+  const [containerWidth, setContainerWidth] = useState(0);
+  const [isMobile, setIsMobile] = useState(false);
+  const containerRef = useRef<HTMLDivElement | null>(null);
 
-  useEffect(() => {
-    const timer = setTimeout(() => {
-      calculateTranslatePercent();
-    }, 100);
+  const aspectRatio = 16 / 9;
+  const desktopMainImageWidth = 980;
```

Comment:
이런 값들은 리렌더링될때 다시 생성될텐데 굳이 그럴 필요가 없죠?

이런 성격의 상수는 컴포넌트 바깥으로 빼서 정의하는게 좋습니다.

Reply:
상수같은 경우는 대문자 + 언더바의 조합으로 정의하는게 관례입니다.

```suggestion
  const DESKTOP_MAIN_IMAGE_WIDTH = 980;
```